### ISSUE TITLE
descpb.TableDescriptor_Reference has reversed meaning for views and sequences

### ISSUE THREAD

**Issue by Xiang-Gu** [cockroachdb/cockroach#92863]
When a table/view/sequence is referenced by another table/view, we store that reference in the
`DependedOnBy` field, which is a slice of `TableDescriptor_Reference` types. In that reference
type, we further track the index ID and/or column ID of the reference, but the direction is reversed
for views and sequences.

In short, if `table_A` is depended on by `table_B`, then 
  - if `table_A` is a view, then `table_A.DependedOnBy` contains a reference whose `indexID` and `columnIDs` are IDs in `table_A`;
  - if `table_A` is a sequence, then `table_A.DependedOnBy` contains a reference whose `indexID` and `columnIDs` are IDs in `table_B`;
 
A concrete example is
```
create sequence s;
create table t (i int primary key default (nextval('s')), j int, index(j));
create view v as select j from t;
create view v2 as select j from t@t_j_idx;
create function f() returns int language sql as $$ select i from t $$;

select crdb_internal.pb_to_json('desc', descriptor)->'table'->'dependedOnBy' from system.descriptor where id = 't'::regclass;
[{"columnIds": [2], "id": 114}, {"columnIds": [2], "id": 115, "indexId": 2}, {"columnIds": [1], "id": 116}]

select crdb_internal.pb_to_json('desc', descriptor)->'table'->'dependedOnBy' from system.descriptor where id = 's'::regclass;
[{"byId": true, "columnIds": [1], "id": 113}]
```
Note how the `columnIds` field means ids in the referenced table (if a view is depended on by), and this field is used to mean ids in the referencing table (if a sequence is depended on by).

**Comment by postamar** [comment]
Yes, this is annoying. We'll fix this opportunistically at some point.
